### PR TITLE
[UD][OPS-11830] Expanded name matching failure responses, fixed redir…

### PR DESCRIPTION
**Pull request by Ufuoma-Daniel**
…ect failiing due to journey not being marked as finished

### CODE REVIEW

**Review by pawelpanasewicz — APPROVED** [review]

File: app/models/namematching/NameMatchingResponse.scala [hunk 1]
```diff
@@ -24,12 +24,15 @@ sealed class SuccessfulMatch extends NameMatchingResponse {
   val isSuccess = true
 }
 
+sealed class FailedMatch extends NameMatchingResponse {
```
Comment (barnesjake) [inline app/models/namematching/NameMatchingResponse.scala:27]: can you make this a sealed trait instead of a class please

Comment (pawelpanasewicz) [reply app/models/namematching/NameMatchingResponse.scala:27]: Sorry, i merged PR before I read this :(

Comment (barnesjake) [reply app/models/namematching/NameMatchingResponse.scala:27]: I was just reading online and there's no functional difference, but just not normal convention (as far as I know)

Comment (Ufuoma-Daniel) [reply app/models/namematching/NameMatchingResponse.scala:27]: NameMatchingResponse is a sealed trait and they cant extend another sealed trait, that's why I just swapped to a class afterward.

**Pull request merged**